crates/rpc/src/pathfinder/methods/get_proof.rs

```diff
@@ -278,6 +298,95 @@ pub async fn get_proof(
 }
 
+/// Returns all the necessary data to trustlessly verify class changes for a
+/// particular contract.
+pub async fn get_proof_class(
+    context: RpcContext,
+    input: GetProofInputClass,
+) -> Result<GetProofOutputClass, GetProofError> {
+    const MAX_KEYS: usize = 100;
+    if input.keys.len() > MAX_KEYS {
+        return Err(GetProofError::ProofLimitExceeded {
+            limit: MAX_KEYS as u32,
+            requested: input.keys.len() as u32,
+        });
+    }
+
+    let block_id = match input.block_id {
+        BlockId::Pending => {
+            return Err(GetProofError::Internal(anyhow!(
+                "'pending' is not currently supported by this method!"
+            )))
+        }
+        other => other.try_into().expect("Only pending cast should fail"),
+    };
+
+    let storage = context.storage.clone();
+    let span = tracing::Span::current();
+
+    let jh = tokio::task::spawn_blocking(move || {
+        let _g = span.enter();
+        let mut db = storage
+            .connection()
+            .context("Opening database connection")?;
+
+        let tx = db.transaction().context("Creating database transaction")?;
+
+        // Use internal error to indicate that the process of querying for a particular
+        // block failed, which is not the same as being sure that the block is
+        // not in the db.
+        let header = tx
+            .block_header(block_id)
+            .context("Fetching block header")?
+            .ok_or(GetProofError::BlockNotFound)?;
+
+        let class_commitment = match header.class_commitment {
+            ClassCommitment::ZERO => None,
+            other => Some(other),
+        };
+
+        // Generate a proof for this class. If the class does not exist, this will
+        // be a "non membership" proof.
+        let class_proof = ClassCommitmentTree::get_proof(&tx, header.number, input.class_hash)
+            .context("Creating contract proof")?
+            .ok_or(GetProofError::ProofMissing)?;
+        let class_proof = ProofNodes(class_proof);
+
+        let class_root_exists = tx
+            .class_root_exists(header.number)
+            .context("Fetching class root existence")?;
+
+        if !class_root_exists {
+            return Ok(GetProofOutputClass {
+                class_commitment,
+                class_proof,
+            });
+        };
+
+        let mut class_proofs = Vec::new();
+        for k in &input.keys {
+            let proof = ClassStorageTree::get_proof(&tx, header.number, k.view_bits())
+                .context("Get proof from class tree")?
+                .ok_or_else(|| {
+                    let e = anyhow!(
+                        "Storage proof missing for key {:?}, but should be present",
+                        k
+                    );
+                    tracing::warn!("{e}");
+                    e
+                })?;
+            class_proofs.push(ProofNodes(proof));
+        }
```

Comment:
Concluding the point on the `keys` param, you never use `class_proofs`. Meaning, we can safely delete the `keys` parameter as suggested above.